src/interface.cpp

```diff
@@ -0,0 +1,166 @@
+#include "interface.h"
+
+interface::interface()
+{
+
+}
+
+void interface::read_gcode_file(std::string filename){
+
+enum g_type {
+    G0=0,
+    G1=1,
+    G2=2,
+    G3=3
+};
+
+//https://github.com/dillonhuff/gpr/
+using namespace gpr;
+
+std::ifstream t(filename.c_str());
+std::string file_contents((std::istreambuf_iterator<char>(t)),
+                          std::istreambuf_iterator<char>());
+
+gcode_program p = parse_gcode(file_contents);
+
+for(int i=0; i<p.num_blocks(); i++){
+    //std::cout<<"gcode line chunck size:"<<p.get_block(i).size()<<std::endl; // Text editor line +1.
+
+    for(int chunk=0; chunk<p.get_block(i).size(); chunk++){
+
+        /*
```

Comment:
Could you remove this commented out block of code?

Reply:
That is nice reference (debug) code. I would prefer it stay's there for info.